lib/platform/azure/index.ts

```diff
@@ -575,9 +575,53 @@ export async function setBranchStatus({
   logger.trace(`Created commit status of ${state} on branch ${branchName}`);
 }
 
-export function mergePr(pr: number, branchName: string): Promise<boolean> {
-  logger.debug(`mergePr(pr)(${pr}) - Not supported by Azure DevOps (yet!)`);
-  return Promise.resolve(false);
+export async function mergePr(
+  pullRequestId: number,
+  branchName: string
+): Promise<boolean> {
+  logger.debug(`mergePr(${pullRequestId}, ${branchName})`);
+  const azureApiGit = await azureApi.gitApi();
+
+  const pr = await azureApiGit.getPullRequestById(
+    pullRequestId,
+    config.project
+  );
+  const mergeMethod = await azureHelper.getMergeMethod(
+    config.repoId,
+    config.project,
+    pr.targetRefName
+  );
```

Comment:
Some caching, needs `config` type changes:

```suggestion
  const mergeMethod = config.mergeMethods[pr.targetRefName] ?? (config.mergeMethods[pr.targetRefName] = await azureHelper.getMergeMethod(
    config.repoId,
    config.project,
    pr.targetRefName
  ));
```

This is what i had in mind with caching the mergeMethods results.